projects/DegenHive/index.js

```diff
@@ -0,0 +1,74 @@
+const sui = require("../helper/chain/sui");
+const { GraphQLClient } = require("graphql-request");
+  
+const graphQLClientMainnet = new GraphQLClient(
+    "https://5lox8etck8.execute-api.eu-central-1.amazonaws.com/api/v1"
+  );
+
+const GET_SUI_POOLS = `query GetSuiPools {
+  getSuiPools {
+    pools {
+      poolId
+    }
+  }
+}`
+
+const GET_SUI_MEME_POOLS = `query GetSuiMemePools {
+  getSuiMemePools {
+    pools {
+      meme_pool_addr
+    }
+  }
+}`
+
+ 
+
+async function tvl(api) {
+    const sui_pools = await graphQLClientMainnet.request(GET_SUI_POOLS, {})
```

Comment:
can you use `cachedGraphQuery` method here, this way, we would be able to compute tvl even if the graph becomes unreachable

Reply:
done